lib/log.h

```diff
@@ -85,6 +85,11 @@ extern void zlog_info(const char *format, ...) PRINTF_ATTRIBUTE(1, 2);
 extern void zlog_notice(const char *format, ...) PRINTF_ATTRIBUTE(1, 2);
 extern void zlog_debug(const char *format, ...) PRINTF_ATTRIBUTE(1, 2);
 
+/* For logs which have error codes associated with them */
+#define zlog_ferr(ferr_id, format, ...)                                        \
+	zlog_err("[EC %d] " format, ferr_id, ##__VA_ARGS__)
```

Comment:
This may be opening a can of worms, but it does seem like RFC 5424 is starting to gain some traction with the cloud log importers and it's supported in syslog-ng and rsyslog. I'm wondering if using something closer to the structured data (https://tools.ietf.org/html/rfc5424#section-6.3) would be nice here.  Something like: [ frr@1234 EC=%d ], or maybe something simpler and non-compliant, but closer (I think key=value is really the main thing).  Splunk likes it and fluentd has RFC 5424 support now.  I realize it's a ways from full RFC 5424 support, but may make the transition easier if we end up going that way...

Reply:
This is a bit orthogonal to what we are doing here.  I'm not saying it's not important either, but I don't see how this would break someone wanting to extend functionality for this in the future.
Given the timeline of getting this done.  I do not think we should be adding additional functionality on top of this